### PR TITLE
RJD-1334/fix_longitudinal_distance

### PR DESCRIPTION
**Pull request by robomic**
## Abstract
This pull requests fixes issue described [here](https://github.com/tier4/scenario_simulator_v2/issues/1364)

## Details

1. Simplification of `HdMapUtils::getLongitudinalDistance` function logic, while retaining its functionalify.
2. Additional matching: previously matching occurred at the starts of the lanelets, now matching is also performed at `0.5 * min(from_lanelet, to_lanelet)` to cover the corner case (please see the mentioned issue description).
3. Additional unit tests for the modified function.
4. During development i have decided that using the _first_ instead of the _second_ approach is far superior (refer to this [discussion](https://github.com/tier4/scenario_simulator_v2/issues/1364#issuecomment-2377464040))
- This approach produces cleaner code in general.
- This approach does not require modification of the underlying centerline's `math::geometry::CatmullRomSpline`.
- The working POC of the first approach is still implemented in the first [commit](https://github.com/tier4/scenario_simulator_v2/pull/1403/commits/bbc98c27b83c88d85ff31f7e61f4e1f665b49164) of this PR.
5. Docs have been updated accordingly.
6. Some recurring scenario conditions have been updated. Currently, majority of these conditions are too precise, this disallows any changes in the calculations.
## References

https://github.com/tier4/scenario_simulator_v2/issues/1364
[Jira ticket](https://tier4.atlassian.net/browse/RJD-1334)

### CODE REVIEW

**Comment by robomic** [comment]
[Regression tests](https://tier4.atlassian.net/browse/RJD-1334?focusedCommentId=191766): OK

File: simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp [hunk 1]
```diff
@@ -1532,85 +1532,71 @@ auto HdMapUtils::getLateralDistance(
...
-      } else {
-        traffic_simulator_msgs::msg::LaneletPose current_lanelet_pose = next_lanelet_pose;
-        current_lanelet_pose.lanelet_id = route[i];
-        if (
-          auto current_lanelet_origin_from_next_lanelet =
-            toLaneletPose(toMapPose(current_lanelet_pose).pose, route[i + 1], 10.0)) {
-          distance -= current_lanelet_origin_from_next_lanelet->s;
+  if (const auto route = getRoute(from_pose.lanelet_id, to_pose.lanelet_id, allow_lane_change);
+      not route.empty() || from_pose.lanelet_id == to_pose.lanelet_id) {
+    static constexpr double matching_distance = 10.0;
```
Comment (hakuturu583) [inline simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp:1550]: Please use the @note comment to explain why the hard coded parameter is the number it is.
Even if there is no specific reason, be sure to leave a comment to that effect.

**Review by hakuturu583 — CHANGES_REQUESTED** [review]
Code is looks good to me and I confirmed this pull request has no-regression.
Please add one comment addition.

File: simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp (hunk 1 above)
Comment (robomic) [reply simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp:1551]: added

**Review by hakuturu583 — APPROVED** [review]
